src/config/constants.ts

```diff
@@ -129,8 +129,9 @@ export const PATTERNS = {
     STRING: /[A-Za-z0-9]+$/,
     APP_NAME: '(?:[a-z0-9]+(?:[.-][a-z0-9]+)*/)*[a-z0-9]+(?:[._-][a-z0-9]+)*',
     CD_PIPELINE_NAME: `^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$`,
-    CONFIG_MAP_KEY: /^[-._a-zA-Z0-9]+$/,
-    SECRET_KEY: /^[-._a-zA-Z0-9]+$/,
+    CONFIG_MAP_AND_SECRET_KEY: /^[-._a-zA-Z0-9]+$/,
```

Comment:
this check should be wrong.. even though kubernetes mentions the same.... please check if we can start key name with -  .  _

Reply:
regex checked